mm2src/coins/utxo/utxo_common.rs

```diff
@@ -1589,10 +1585,11 @@ pub fn send_taker_refunds_payment<T: UtxoCommonOps + SwapOps>(
 
 pub fn send_taker_payment_refund_preimage<T: UtxoCommonOps + SwapOps>(
     coin: T,
-    taker_payment_refund_preimage: &[u8],
+    watcher_refunds_payment_args: SendWatcherRefundsPaymentArgs,
 ) -> TransactionFut {
-    let transaction: UtxoTx =
-        try_tx_fus!(deserialize(taker_payment_refund_preimage).map_err(|e| TransactionErr::Plain(format!("{:?}", e))));
+    let transaction: UtxoTx = try_tx_fus!(
```

Comment:
What do you think about taking `T` as ref here and cloning once inside the function instead of cloning everywhere the function is used? 

```rust
ub fn send_taker_payment_refund_preimage<T: UtxoCommonOps + SwapOps>(
     coin: &T,
     watcher_refunds_payment_args: SendWatcherRefundsPaymentArgs,
 ) -> TransactionFut {
let coin = coin.clone()
....
}
```

Reply:
Yes, this is better